### PR TITLE
Dockerize Level 1 Onboarding

### PR DESCRIPTION
**Pull request by farzanmirshekari**
# Purpose
PR for [this](https://www.notion.so/uworbital/Improve-dev-env-setup-for-level1-challenge-7b494c39e0934b23aeff297c255741bf?pvs=4)

# New Changes
Dockerfile added, using Ubuntu, GCC and Make. Updated README.md with new instructions.

# Testing
Able to build container and run it locally.

# Outstanding Changes
N/a.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
+```
 
-To test your solution, run:
-```sh
-make clean
-make all
-./build/challenge # Or ./build/challenge.exe on Windows
+To run the tests, run:
+```
+docker run --rm -it fw-onboarding-level1
```
Comment (dgobalak) [inline README.md]: Can you write a Notion doc on installing and setting up Docker and link that here? I'd assume most new members won't already have Docker.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM ubuntu:latest
+
+WORKDIR /app
+
+RUN apt-get update && apt-get install -y gcc make
+
+COPY . /app
```
Comment (dgobalak) [inline Dockerfile]: If someone made changes to their solution code and wanted to try running it again, wouldn't they have to rebuild the image to copy over the updated file? It'd be nice if they didn't have to do that.

**Review by dgobalak — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (farzanmirshekari) [reply Dockerfile]: Done, they can now use the bash shell in the container to compile/run their solutions.

File: README.md (hunk 1 above)
Comment (farzanmirshekari) [reply README.md]: Notion doc added [here](https://www.notion.so/uworbital/How-To-Docker). It's a fairly quick introduction to understand the commands used in the Dockerfile, plus a general introduction and overview of Docker. Let me know if there is anything missing/you want added there.

Comment (farzanmirshekari) [reply README.md]: I can't seem to edit the [Firmware Onboarding](https://www.notion.so/uworbital/Firmware-Onboarding) page to add a link to the How-To though.

Comment (Navtajh04) [reply README.md]: you should be able to now

Comment (farzanmirshekari) [reply README.md]: Yup, added it to the list of How-Tos as well.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
```
Comment (dgobalak) [inline Dockerfile]: This COPY isn't needed anymore because we're mounting the directory

**Review by dgobalak — CHANGES_REQUESTED** [review]
final thing

File: Dockerfile (hunk 3 above)
Comment (farzanmirshekari) [reply Dockerfile]: Yup, should be good now!

**Review by dgobalak — APPROVED** [review]
Lgtm

**Pull request merged**